File: KeepNoteAPIs/NoteService/NoteService.API/Models/Note.cs
using System;
using System.Collections.Generic;
using MongoDB.Bson.Serialization.Attributes;

namespace NoteService.API.Models
{
    public class Note
    {
        [BsonId]
        public int Id { get; set; }
        public string Title { get; set; }
        public string Content { get; set; }
        public DateTime CreationDate { get; set; }
        public string CreatedBy { get; set; }
        public Category Category { get; set; }
        public List<Reminder> Reminders { get; set; }
        public List<string> Tags { get; set; }
    }
}

File: ClientApp/src/app/models/note.ts
import { Tag } from "./tag";
import { Reminder } from "./reminder";

export class Note {
  id: Number;
  title: string;
  content: string;
  state: string;
  userid: Number;
  tags: Array<string>;
  category: Tag;
  reminders: Array<Reminder>;

  constructor() {
    this.category = new Tag();
    this.reminders = Array<Reminder>();
    this.title = '';
    this.content = '';
    this.state = 'not-started'; // default state
  }
}


export class NoteMapper {
  id: Number;
  title: string;
  content: string;
  userid: Number;
  reminders: Array<Reminder>;
  category: Tag;
  
  constructor() {
    this.title = '';
    this.content = '';
  }
}